btree/btree.c

```diff
@@ -2921,6 +2939,19 @@ static void ff_capture(struct slot *slot, struct m0_be_tx *tx)
 	M0_BTREE_TX_CAPTURE(tx, seg, h, hsize);
 }
 
+static void ff_node_free_credits(const struct nd *node,
+				 struct m0_be_tx_credit *accum)
+{
+	struct ff_head *h           = ff_data(node);
+	int             shift       = h->ff_shift;
+	int             node_size   = 1ULL << shift;
+	int             header_size = sizeof(*h);
+
+	m0_be_allocator_credit(NULL, M0_BAO_FREE_ALIGNED,
```

Comment:
ff_node_free_credits() should return only the size that you want to add to your credits.
Actual addition of credits should be done outside of this function, i.e. in btree_del_credit()
Otherwise for every node_type() you will be calling an m0_be_allocator_credit() and m0_be_tx_credit_add() which is not good.
With this change, the function name will also require some modification.

Reply:
@ydb242 The credit addition happens in the credit functions in the old btree code. They seem to take the last parameter 'accum' and add the credits which they need to the passed accum structure.